src/main/java/uk/gov/justice/digital/zone/StructuredZoneCDC.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.justice.digital.zone;
+
+import lombok.val;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.justice.digital.config.JobArguments;
+import uk.gov.justice.digital.converter.dms.DMS_3_4_6;
+import uk.gov.justice.digital.domain.model.SourceReference;
+import uk.gov.justice.digital.exception.DataStorageException;
+import uk.gov.justice.digital.service.DataStorageService;
+import uk.gov.justice.digital.service.SourceReferenceService;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.function.Consumer;
+
+import static org.apache.spark.sql.functions.col;
+import static uk.gov.justice.digital.converter.dms.DMS_3_4_6.Operation.*;
+import static uk.gov.justice.digital.converter.dms.DMS_3_4_6.ParsedDataFields.OPERATION;
+import static uk.gov.justice.digital.zone.RawZone.PRIMARY_KEY_NAME;
+
+public class StructuredZoneCDC extends StructuredZone implements DeltaWriter {
+
+    private static final Logger logger = LoggerFactory.getLogger(StructuredZone.class);
+
+    @Inject
+    public StructuredZoneCDC(
+            JobArguments jobArguments,
+            DataStorageService storage,
+            SourceReferenceService sourceReference
+    ) {
+        super(jobArguments, storage, sourceReference);
+    }
+
+    @Override
+    public Dataset<Row> process(SparkSession spark, Dataset<Row> dataFrame, Row table) throws DataStorageException {
+        val filteredRecords = dataFrame.filter(col(OPERATION).isin(cdcOperations));
+        return super.process(spark, filteredRecords, table);
+    }
+
+    @Override
+    public void writeValidRecords(
```

Comment:
As above, ideally this should not be public since it's only really `process` that's part of the public API here.